### PR TITLE
Create new container new ci dependencies (v47)

### PR DESCRIPTION
**Pull request by andreitraistaru**
## Reason

Adding required dependencies for further improvements in testing tools.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] New `unsafe` code is documented.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md

### CODE REVIEW

File: tools/devctr/pyproject.toml [hunk 1]
```diff
@@ -5,22 +5,28 @@ description = "Firecracker python dependencies"
 authors = ["Your Name <you@example.com>"]
 
 [tool.poetry.dependencies]
+aws-embedded-metrics = "*"
 black = "*"
 boto3 = "*"
 botocore = "*"
 dataclasses = "*"
 gitlint = "0.16.0"
+less = "*"
```
Comment (pb8o) [inline tools/devctr/pyproject.toml:14]: This should go in the Dockerfile / apt part. Funny there is also a less in PyPi.

File: tools/devctr/pyproject.toml [hunk 2]
```diff
@@ -5,22 +5,28 @@ description = "Firecracker python dependencies"
...
+pytest-xdist = "*"
 python = "3.10.4"
 PyYAML = "*"
 requests = "*"
 requests-unixsocket = "*"
 retry = "*"
+setproctitle = "*"
 typing-extensions = "*"
 # We need to fix the gitlint version since version 0.17.0 has a problem:
 # https://pythonissues.com/issues/1410824
```
Comment (pb8o) [inline tools/devctr/pyproject.toml:32]: I know it's not introduced in your change, but could you move the comment to the gitlint line?

**Review by pb8o — CHANGES_REQUESTED** [review]

File: tools/devctr/pyproject.toml (hunk 1 above)
Comment (andreitraistaru) [reply tools/devctr/pyproject.toml:14]: oh... yeah... that's right... it's the tool, not the dependency what we need; fixing that

File: tools/devctr/pyproject.toml (hunk 2 above)
Comment (andreitraistaru) [reply tools/devctr/pyproject.toml:32]: yes, sure, I'll move it before the gitlint line for better visibility

File: tools/devctr/pyproject.toml (hunk 1 above)
Comment (andreitraistaru) [reply tools/devctr/pyproject.toml]: I fixed that (and also updated the docker image).

**Review by pb8o — APPROVED** [review]

**Review by mattschlebusch — APPROVED** [review]

**Pull request merged**